### ISSUE TITLE
video recording loses audio upon export

### ISSUE THREAD

**Issue by harveyB** [BradLarson/GPUImage#1882]
hi,

we are seeing a problem in iOS8 where the recorded video, when exported to camera roll using standard code is losing audio - the video is recorded using GPUMovieWriter as mp4 using all standard settings - no change is made to tracks or composition or doing anything to the video - we are not even using avexportsession - simply using ALAssetsLibrary writeVideoAtPathToSavedPhotosAlbum

- loses audio upon export to camera roll
- it plays back with audio in our app using MPMoviePlayerController
- it continues to work in iOS7
- further, when we export to desktop and load in quicktime, it doesn't have audio track
- but if we open with VLC, it shows audio track and plays audio.

so it seems something changed with iOS8, with apple mp4 video audio format/composition/encoding/playback that gpumoviewriter needs to account for.
- is anyone else having this issue?

thx.

**Comment by cyoub** [comment]
Yes we are having the same issue. Audio Export fails in iOS 8, but works fine in iOS 7. Could you fix the problem on your end ?

**Comment by adouggy** [comment]
we face same thing, iOS9 no problem, but iOS8 sometimes lost audio
